platform/mds/mds/src/main/java/org/motechproject/mds/builder/impl/EntityMetadataBuilderImpl.java

```diff
@@ -134,8 +139,10 @@ public void addHelperClassMetadata(JDOMetadata jdoMetadata, ClassData classData,
         cmd.setIdentityType(IdentityType.APPLICATION);
         cmd.setPersistenceModifier(ClassPersistenceModifier.PERSISTENCE_CAPABLE);
 
-        InheritanceMetadata imd = cmd.newInheritanceMetadata();
-        imd.setCustomStrategy("complete-table");
+        if(entityType == EntityType.STANDARD)
+            fixDiscriminator(definition, cmd, 1);
```

Comment:
In this class, 'fix' is a term we applied to methods that work on the metadata after the enhancer runs. This is executed during regular generation, so I would start the name with 'generate' or something along those lines.